query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PrometheusBasedRequestHandler.java

```diff
@@ -81,6 +93,35 @@ public QueryCost canHandle(QueryRequest request, ExecutionContext executionConte
 
     // todo call convert and execute request using client here
 
-    return null;
+    Map<Request, PromQLMetricResponse> responseMap;
+    Map<String, String> metricNameToQueryMap;
+    if (isRangeQueryRequest(originalRequest)) {
+      PromQLRangeQueries promQLRangeQueries =
+          requestToPromqlConverter.convertToPromqlRangeQuery(
+              executionContext, originalRequest, executionContext.getAllSelections());
+      metricNameToQueryMap = promQLRangeQueries.getMetricNameToQueryMap();
+      responseMap = prometheusRestClient.executeRangeQuery(promQLRangeQueries);
+    } else {
+      PromQLInstantQueries promQLInstantQueries =
+          requestToPromqlConverter.convertToPromqlInstantQuery(
+              executionContext, originalRequest, executionContext.getAllSelections());
+      metricNameToQueryMap = promQLInstantQueries.getMetricNameToQueryMap();
```

Comment:
If, we are going to change the `promQLInstantQueries` and `PromQLRangeQueries` with map<> internally, we will not need this input to builder.